addons/project_timesheet_holidays/models/hr_holidays.py

```diff
@@ -16,29 +16,42 @@ def _default_task_id(self):
         company = self.company_id if self.company_id else self.env.company
         return company.leave_timesheet_task_id.id
 
-    timesheet_generate = fields.Boolean('Generate Timesheet', default=True, help="If checked, when validating a time off, timesheet will be generated in the Vacation Project of the company.")
+    timesheet_generate = fields.Boolean(
+        'Generate Timesheet',
+        compute='_compute_timesheet_generate',
+        store=True,
+        readonly=False,
+        copy=True,
+        help="If checked, when validating a time off, timesheet"
+             " will be generated in the Vacation Project of the company.")
     timesheet_project_id = fields.Many2one('project.project', string="Project", default=_default_project_id, domain="[('company_id', '=', company_id)]", help="The project will contain the timesheet generated when a time off is validated.")
-    timesheet_task_id = fields.Many2one('project.task', string="Task for timesheet", default=_default_task_id, domain="[('project_id', '=', timesheet_project_id), ('company_id', '=', company_id)]")
-
-    @api.onchange('timesheet_task_id')
-    def _onchange_timesheet_generate(self):
-        if self.timesheet_task_id or self.timesheet_project_id:
-            self.timesheet_generate = True
-        else:
-            self.timesheet_generate = False
-
-    @api.onchange('timesheet_project_id')
-    def _onchange_timesheet_project(self):
-        company = self.company_id if self.company_id else self.env.company
-        default_task_id = company.leave_timesheet_task_id
-        if default_task_id and default_task_id.project_id == self.timesheet_project_id:
-            self.timesheet_task_id = default_task_id
-        else:
-            self.timesheet_task_id = False
-        if self.timesheet_project_id:
-            self.timesheet_generate = True
-        else:
-            self.timesheet_generate = False
+    timesheet_task_id = fields.Many2one(
+        'project.task',
+        string="Task for timesheet",
+        compute='_compute_timesheet_task_id',
+        store=True,
+        readonly=False,
+        default=_default_task_id,
+        domain="[('project_id', '=', timesheet_project_id),"
+                "('company_id', '=', company_id)]")
+
+    @api.depends('timesheet_task_id', 'timesheet_project_id')
+    def _compute_timesheet_generate(self):
+        for holiday_type in self:
+            holiday_type.timesheet_generate = holiday_type.timesheet_task_id and \
+                holiday_type.timesheet_project_id
+
+    @api.depends('timesheet_project_id')
+    def _compute_timesheet_task_id(self):
+        for record in self:
```

Comment:
`for leave_type in self`

Reply:
already done in the current version.

Reply:
same here IIRC, changed in two different commits